src/OrchardCore.Modules/OrchardCore.Media/Services/AttachedMediaFieldFileService.cs

```diff
@@ -80,9 +84,17 @@ private void RemoveTemporary(List<EditMediaFieldItemInfo> items)
         // Newly added files
         private void MoveNewFilesToContentItemDirAndUpdatePaths(List<EditMediaFieldItemInfo> items, ContentItem contentItem)
         {
-            items.Where(x => !x.IsRemoved).ToList()
+            items.Where(x => !x.IsRemoved && !string.IsNullOrEmpty(x.Path)).ToList()
                 .ForEach(async x =>
                 {
+                    var fileInfo = await _fileStore.GetFileInfoAsync(x.Path);
+
+                    if (fileInfo == null)
+                    {
+                        _logger.LogError("A file with the path '{x.path}' does not exist.");
```

Comment:
This is not how to log structured data. It should be `{Path}` and the value passed as a second parameter. Log for other examples in the code base.